audio_service/android/src/main/java/com/ryanheise/audioservice/AudioService.java

```diff
@@ -638,30 +695,43 @@ void playMediaItem(MediaDescriptionCompat description) {
     /**
      * Updates metadata, loads the art and updates the notification.
      * Gets called from background thread.
-     *
+     * <p>
      * Also adds the loaded art bitmap to the MediaMetadata.
      * This is needed to display art in lock screen in versions
      * prior Android 11, in which this feature was removed.
-     *
+     * <p>
      * See:
      *  - https://developer.android.com/guide/topics/media-apps/working-with-a-media-session#album_artwork
      *  - https://9to5google.com/2020/08/02/android-11-lockscreen-art/
      */
     synchronized void setMetadata(MediaMetadataCompat mediaMetadata) {
         String artCacheFilePath = mediaMetadata.getString("artCacheFile");
         if (artCacheFilePath != null) {
-            artBitmap = loadArtBitmapFromFile(artCacheFilePath);
-            mediaMetadata = new MediaMetadataCompat.Builder(mediaMetadata)
-                    .putBitmap(MediaMetadataCompat.METADATA_KEY_ALBUM_ART, artBitmap)
-                    .putBitmap(MediaMetadataCompat.METADATA_KEY_DISPLAY_ICON, artBitmap)
-                    .build();
+            // Load local files and network images, cached in files
+            artBitmap = loadArtBitmap(artCacheFilePath, null);
+            mediaMetadata = putArtToMetadata(mediaMetadata);
+        } else {
+            // Load content:// URIs
+            String artUri = mediaMetadata.getString(MediaMetadataCompat.METADATA_KEY_DISPLAY_ICON_URI);
+            if (artUri != null) {
```

Comment:
What if artCacheFilePath is null and artUri is also null? Shouldn't artBitmap be set to null?

Reply:
Weirdly it doesn't change anything, but yes, that would be a correct logic - fixed it